osu.Framework/IO/Network/WebRequest.cs

```diff
@@ -486,15 +488,21 @@ protected virtual void Complete(Exception e = null)
             else
                 logger.Add($@"Request to {Url} successfully completed!");
 
+            // if a failure happened on performing the request, there are still situations where we want to process the response.
+            // consider the case of a server returned error code which triggers a WebException, but the server is also returning details on the error in the response.
             try
             {
                 if (!wasTimeout)
                     ProcessResponse();
             }
             catch (Exception se)
             {
-                logger.Add($"Processing response from {Url} failed with {se}.");
-                e = e == null ? se : new AggregateException(e, se);
+                // that said, we don't really care about an error when processing the response if there is already a higher level exception.
+                if (e != null)
+                {
+                    logger.Add($"Processing response from {Url} failed with {se}.");
+                    e = se;
```

Comment:
Upon reflection, not sure about this one... Lower down this does `throw e;`, which means that the stack trace of `se` will get discarded by doing this (previously the aggregate was wrapping it; with this change, the exception will always appear to originate from this line). Maybe this should just be a

```suggestion
                    throw;
```

if we don't care what gets thrown anyway?

Reply:
If you check back in history, we are only processing the response for a sepcific error handling case in osu-web. I really don't think we want aggregate exceptions to be fired for this because it just confuses the situation with no gain.

I'm not sure what you mean by the stack trace loss. Are you saying that by assigning and exception it is changing the stack trace somehow?

If this is really important then maybe changing the completion conditions to a `finally` block and using `throw` could work.

Reply:
As far as I'm aware `throw` and `throw ex` are semantically different in a catch block ([reference SO post](https://stackoverflow.com/a/730255), [reference sharplab](https://sharplab.io/#v2:C4LglgNgPgAgTARgLACh4AZXwMwAIDequxu8cBRJVMCAnABQBEgPBuAg+4wJQDclxAvr1yCADgCcwANwCGwAKakEANlIAWXABVZAZ2ABBejXJSAxsDAB7AHYdBhFFSrBRAT0FU7Dz7hNmrAOgBJSwlzAGtZem43EgF7L2MZYwALXHoAUQAPY1lhX0tcWQybOM8PL0ck0XMAdwKMnhKHWKpmkhFxaTkFZRg1TR0AIQMEI1MLa1to4idXRvcphx9xwOCwiKi5mIWE4GTUzOzc8bri8rLy6cqahvLW/lQ+IA===)). From what I understand `throw` without the `ex` does some magic to capture the previous exception's stack trace and forward it, while `throw ex` just acts like a throw from its immediate position in code (potentially obfuscating where `ex` was originally coming from).

Reply:
The exception we'd care about a location of is more likely the `e` not the `se`. I've change to a throw anyways.